### PR TITLE
Wrap SOME of Radix Engine Toolkit types, first of MANY PRs.

### PR DESCRIPTION
**Pull request by CyonAlexRDX**
# LOC Delta
Fear not the LOC delta, it is the huge generated Swift file mostly.

# Introduction
This PR primary solves the so far unsolved challenge of how to implement and vendor `Decimal`, renamed to `Decimal192` as to not collide with Swift Foundations `Decimal`. 192 comes from the fact that it is powered by a 192 bit integer inside of Scrypto repo (`radix_engine_common` crate).

I've solved it by ensuring that `Decimal192` is a struct/data class in Swift and Kotlin, and NOT a typealias, which HAS a `__inner: String` which is named like that so to scare Swift/Kotlin devs to not touch it, while keeping it as a thing which has a thing which has a ScryptoDecimal192. This actives the best of two worlds:
1. Strong typing in Swift and Kotlin
2. String typing in Rust.

The goal of this PR is to progress the wrapping of RET inside of Sargon, fulfilling the defined "API" by PR: https://github.com/radixdlt/babylon-wallet-ios/pull/1032

This PR also implements, and stubs out some other RET types, and introduces fundamental building blocks such as `Secp256k1Signature` - UniFFI exposed! and `Ed25519Signature` also UniFFI exposed, needed by `SignedIntent` and other RET types.

This PR also contains improvements to the iOS example app

# Changes

## Changes in Rust land
* Decimal renamed to Decimal192 and made production ready. Missing some formatting methods
* Create Hex64, Hex33, Hex64 bytes types, by changing Hex32Bytes to a macro, this allows typesafe, non generic value (due to restrictions in UniFFI) types for bytes, which we can use in Signature types
* Secp256k1Signature and Ed25519Signature types
* Nonce
* Epoch
* Dummy placeholder RET types
* Wallet now vendors ` mnemonic_with_passphrase_of_device_factor_source_by_id` which is used by Example wallet to load the mnemonic

## Changes in Sargon Swift SPM Package
* Swift-ify some types in SPM target `sargon` mostly wrapping freestanding global functions in methods by use of extensions.
* Make Decimal192 conform to SignedNumeric swift protocol

## Changes in Example wallet iOS
* Make create account a flow, user can select name and appearance ID.
* Add screen to show mnemonic

# Future
Many more PRs to come, all focused on **vendoring RET through Sargon**..

### CODE REVIEW

File: src/sargon.udl [hunk 1]
```diff
@@ -10,4 +10,7 @@ typedef string Timestamp;
 [Custom]
 typedef sequence<i8> BagOfBytes;
 
+[Custom]
+typedef string InnerDecimal;
```
Comment (CyonAlexRDX) [inline src/sargon.udl:14]: Here is the trick, we vendor `Decimal192` as a struct with one single stored property (field) `__inner: InnerDecimal` which in the land of Swift and Kotlin will be `__inner: String`. It is given that name to hide it as much as possible. It is, in fact, NOT a String but the rust type `InnerDecimal` which in Rust land holds a ScryptoDecimal192. 

This is the best compromise, it allwos for usage of a real decimal in Rust land - the ScryptoDecimal192 - and it allows for its OWN type in Swift/Kotlin, and NOT a silly error prone `typealias Decimal192 = String`.

File: src/core/types/signatures/signature_with_public_key.rs
Comment (CyonAlexRDX) [inline src/core/types/signatures/signature_with_public_key.rs:1]: To Be Written in coming PR

File: src/wrapped_radix_engine_toolkit/manifest_summary.rs
Comment (CyonAlexRDX) [inline src/wrapped_radix_engine_toolkit/manifest_summary.rs:1]: stubbing out the API as per https://github.com/radixdlt/babylon-wallet-ios/pull/1032

File: src/wrapped_radix_engine_toolkit/notarized_transaction.rs
Comment (CyonAlexRDX) [inline src/wrapped_radix_engine_toolkit/notarized_transaction.rs:1]: stubbing out the API as per https://github.com/radixdlt/babylon-wallet-ios/pull/1032

File: src/wrapped_radix_engine_toolkit/transaction_intent.rs
Comment (CyonAlexRDX) [inline src/wrapped_radix_engine_toolkit/transaction_intent.rs:1]: stubbing out the API as per https://github.com/radixdlt/babylon-wallet-ios/pull/1032

File: src/wrapped_radix_engine_toolkit/transaction_manifest.rs
Comment (CyonAlexRDX) [inline src/wrapped_radix_engine_toolkit/transaction_manifest.rs:1]: stubbing out the API as per https://github.com/radixdlt/babylon-wallet-ios/pull/1032

`blobs()` method will probably be turned into just field usage.

File: src/core/types/decimal192.rs
Comment (CyonAlexRDX) [inline src/core/types/decimal192.rs:1]: Almost ALL methods inits of Decimal as per https://github.com/radixdlt/babylon-wallet-ios/pull/1032 has been implemented, formatting remains.

File: .pre-commit-config.yaml
Comment (CyonAlexRDX) [inline .pre-commit-config.yaml:1]: @micbakos-rdx please re-install the precommit hook

File: src/wrapped_radix_engine_toolkit/transaction_header.rs
Comment (GhenadieVP) [inline src/wrapped_radix_engine_toolkit/transaction_header.rs:1]: 👍

**Review by GhenadieVP — APPROVED** [review]
Awesome, looks promising!
Waiting to see how stuff around TransactionManifest and building will look like 🚀.

**Pull request merged**